internal/ethapi/api.go

```diff
@@ -1010,6 +1010,60 @@ func (s *BlockChainAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rpc.
 	return result, nil
 }
 
+func (s *BlockChainAPI) GetBlobSidecars(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) ([]map[string]interface{}, error) {
+	block, err := s.b.BlockByNumberOrHash(ctx, blockNrOrHash)
+	if block == nil || err != nil {
+		// When the block doesn't exist, the RPC method should return JSON null
+		// as per specification.
+		return nil, nil
+	}
+	blobSidecars, err := s.b.GetBlobSidecars(ctx, block.Hash())
+	if err != nil || len(blobSidecars) == 0 {
+		return nil, nil
+	}
+	result := make([]map[string]interface{}, len(blobSidecars))
+	blobIndex := -1
+	for txIndex, tx := range block.Transactions() {
+		if tx.Type() == types.BlobTxType {
+			blobIndex++
+			if blobIndex >= len(blobSidecars) {
+				return nil, fmt.Errorf("blobSidecars length mismatch %d", len(blobSidecars))
+			}
+			result[blobIndex] = marshalBlobSidecar(blobSidecars[blobIndex], block.Hash(), block.NumberU64(), tx, txIndex)
+		}
+	}
+	return result, nil
+}
+
+func (s *BlockChainAPI) GetBlobSidecarByTxHash(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
+	tx, blockHash, blockNumber, Index := rawdb.ReadTransaction(s.b.ChainDb(), hash)
+	if tx == nil {
+		return nil, nil
+	}
+	block, err := s.b.BlockByHash(ctx, blockHash)
+	if block == nil || err != nil {
+		// When the block doesn't exist, the RPC method should return JSON null
+		// as per specification.
+		return nil, nil
+	}
+	blobSidecars, err := s.b.GetBlobSidecars(ctx, blockHash)
+	if err != nil || len(blobSidecars) == 0 {
+		return nil, nil
+	}
+	blobIndex := -1
+	for txIndex, tx := range block.Transactions() {
+		if tx.Type() == types.BlobTxType {
+			if txIndex <= int(Index) {
+				blobIndex++
+			} else {
+				return nil, fmt.Errorf("blobSidecars length mismatch %d", len(blobSidecars))
+			}
+		}
+	}
+	result := marshalBlobSidecar(blobSidecars[blobIndex], blockHash, blockNumber, block.Transaction(hash), int(Index))
```

Comment:
we don't need call `block.Transaction(hash)` here, as we already have the tx at the beginning of the function, just rename it with txTarget should be ok.